TestFoundation/TestURLSession.swift

```diff
@@ -119,6 +120,56 @@ class TestURLSession : LoopbackServerTest {
         waitForExpectations(timeout: 12)
     }
     
+    func test_dataTaskWithHttpInputStream() {
+        func randomString(length: Int) -> String {
+            let letters = Array("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789")
+            let len = letters.count
+            
+            var randomString = ""
+            
+            for _ in 0 ..< length {
+                let rand = Int.random(in: 0..<len)
+                let nextChar = letters[rand]
+                randomString += String(nextChar)
+            }
+            return randomString
+        }
+        
+        let delegate = HTTPBinResponseDelegateJSON<HTTPBinResponse>()
+
+        let dataString = randomString(length: 65537)
+        
+        let urlString = "http://httpbin.org/post"
```

Comment:
We have a policy of not reaching out to external hosts during unit tests, because these have a tendency to be unreliable in the Swift CI environment. Can you use the [toy HTTP server we have in the tests](https://github.com/apple/swift-corelibs-foundation/blob/4438c6e2676c07cf4e28671bf1cd888ae9e4bc2c/TestFoundation/HTTPServer.swift) instead?